### ISSUE TITLE
Futures / Async support

### ISSUE THREAD

**Issue by sksamuel** [kotlintest/kotlintest#94]
In ScalaTest there's very handy support for Scala futures.

``` scala
val f : Future[Any] = ...
whenReady(f) {
 // assertions here
}
```

I think we should add the same to KotlinTest. However Kotlin currently lacks a standard "Future" implementation. There is of course the stuff in the JDK, but then there's Guava, and various other things as well. 

Once 1.1 drops with async / co-routines, I think we'll find Kotlin comes up to par with what Scala provides, and once that happens we should add testing hooks that make it very easy to work with async / co-routines.

**Comment by jmfayard** [comment]
FYI, I implemented a very simple wrapper around a retrofit service that returned rxjava Single and was very happy with it.

```kotlin
class ApiTests : RxSpec() { init {

    retrofitFeature("GET ping", service.pingServer()) {
        val map = this
        scenario("server ok") {
            map["ok"] shouldBe true
        }
    }
}
```

with `retrofitFeature` being defined like something like this

```kotlin
abstract class RxSpec : FeatureSpec() {
    protected fun <T> retrofitFeature(name: String, call: Single<Response<T>>>, operation: T.() -> Unit) =
        feature(name) {
            val response = call.toBlocking().value()
            var data : T? = null
            if (response.isSuccessful) {
                val body = response.body()
                body?.operation()
            } else {
                scenario("Call $name failed with a server error") {
                    "HTTP ${response.message()}" shouldBe "HTTP 200"
                }
            }
        }

```

**Comment by sksamuel** [comment]
Implemented in master, will be included in 3.0.0 RC2.
